### ISSUE TITLE
[BUG] `TestTabWithColorBox` test fails on macos terminal

### ISSUE THREAD

**Issue by nedpals** [Delta456/box-cli-maker#29]
**Describe the bug**
When running the test especially on `TestTabWithColorBox`, it crashes (specifically an out-of-bounds crash)

**To Reproduce**

1. Test `go test -v .`
2. See error

**Expected behavior**
Successful test

**Screenshots / Logs**
![image](https://user-images.githubusercontent.com/7358345/185941115-4a6aeccc-9398-478c-a127-b2e588c36138.png)

**Versions (please complete the following information, if relevant):**

- Software version: dev branch
- OS: Mac OS 11.6
- Golang version: 1.19
- Terminal using and Version: MacOS Terminal 2.11

**Additional context**
Debugging the text and the separator in runes shows the following results:

```
color.ClearCode(topBar): [9484 32 66 111 120 32 32 32 32 32 67 76 73 32 32 32 32 32 77 97 107 101 114 32 32 32 128230 32 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9472 9488]
color.ClearCode(title): [66 111 120 32 32 32 67 76 73 32 32 32 32 32 77 97 107 101 114 32 32 32 128230]
```

the title text in the topbar in `[]runes` (the `color.clearCode(topBar)`) is `66 111 120 32 32 32 32 32 67 76 73 32 32 32 32 32 77 97 107 101 114 32 32 32 128230`
while the separator (the `color.clearCode(title)`) is `66 111 120 32 32 32 67 76 73 32 32 32 32 32 77 97 107 101 114 32 32 32 128230`

if you look at it closely on the first set of `32`s the text has 5 `32`s while the separator has only 3 `32`s

probably an issue with the `ClearCode` method

**Comment by Delta456** [comment]
This is quite an interesting issue.

**Comment by inhere** [comment]
hi @nedpals can provide the `topBar` and `title` string.

**Comment by Delta456** [comment]
I can confirm that this test works on `iterm2`

![img](https://media.discordapp.net/attachments/840291790657683517/1011930852189548594/Screenshot_2022-08-24_at_3.01.46_PM.png)

**Comment by nedpals** [comment]
> hi @nedpals can provide the `topBar` and `title` string.

here are the values (i'll just paste them as []rune as turning them to strings might lose info)
```go
topBar := []rune{9484, 32, 66, 111, 120, 32, 32, 32, 32, 32, 67, 76, 73, 32, 32, 32, 32, 32, 77, 97, 107, 101, 114, 32, 32, 32, 128230, 32, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9472, 9488}

title := []rune{66, 111, 120, 32, 32, 32, 67, 76, 73, 32, 32, 32, 32, 32, 77, 97, 107, 101, 114, 32, 32, 32, 128230}
```

**Comment by inhere** [comment]
there are `runes` is after call `color.ClearCode()` ?
can provide them on before call func `ClearCode()`

**Comment by Delta456** [comment]
> there are `runes` is after call `color.ClearCode()` ? can provide them on before call func `ClearCode()`

There are the `runes` before `ClearCode` function call.

`color.ClearCode(topBar)` has `[]rune{66, 111, 120, 32, 32, 32, 32, 32, 67, 76, 73, 32, 32, 32, 32, 32, 77, 97, 107, 101, 114, 32, 32, 32, 128230}`

`color.ClearCode(title)` has `[]rune{66, 111, 120, 32, 32, 32, 67, 76, 73, 32, 32, 32, 32, 32, 77, 97, 107, 101, 114, 32, 32, 32, 128230}`

**Comment by Delta456** [comment]
@inhere Can you work on solving this?

**Comment by inhere** [comment]
@Delta456 `xstrings.ExpandTabs` maybe has error

see:

![image](https://user-images.githubusercontent.com/5302062/188313092-3174dee0-32e7-4ed3-ba52-2f2f9b8f104d.png)

**Comment by Delta456** [comment]
> @Delta456 `xstrings.ExpandTabs` maybe has error
> 
> see:
> 
> ![image](https://user-images.githubusercontent.com/5302062/188313092-3174dee0-32e7-4ed3-ba52-2f2f9b8f104d.png)

This seems to work on `iterm2` and the rest terminals except `iterm`. Can you check what should be the tab size for `iterm`? I will add a workaround for it.

**Comment by inhere** [comment]
hi @Delta456  I am running the test directly in MacOS GoLand

**Comment by Delta456** [comment]
> hi @Delta456 I am running the test directly in MacOS GoLand

I see. I really wonder why it fails on `iterm`.

**Comment by griimick** [comment]
I don't think you can even run `iterm` in the newer versions of MacOS. Couldn't repro this issue in MacOS Terminal 2.12.7 or Iterm2 v3.4.16 in MacOS v12.6
